HIP/hip-1195.md

```diff
@@ -0,0 +1,834 @@
+---
+hip: 1195
+title: Hiero hooks and an application to allowances
+author: Michael Tinker <@tinker-michaelj>
+working-group: Richard Bair <@rbair23>, Jasper Potts <@jasperpotts>, Atul Mahamuni <atul@hashgraph.com>,  Matthew DeLorenzo <@littletarzan>, Edward Wertz <@edward-swirldslabs>, Steven Sheehy <@steven-sheehy>
+requested-by: Hashgraph
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+created: 2025-02-19
+discussions-to: https://github.com/hiero-ledger/hiero-improvement-proposals/discussions/1172
+updated: 2025-05-13
+-------------------
+
+## Abstract
+
+We propose **hooks**, programmable Hiero extension points that let users customize the behavior of their entities.
+In principle, hooks could be programmed in any language, but we begin with **EVM hooks**. Users program EVM hooks 
+by writing contracts in a language like Solidity that compiles to EVM bytecode. EVM hooks are either **pure**
```

Comment:
What would happen if an EVM hook attempted to access EVM-specific parameters that are not applicable in a cryptoTransfer context, such as `msg.value`, `tx.origin`, `block.number`, etc.?

I think the HIP should have a dedicated section that clarifies the differences between a smart contract published inside HSCS and as a hook.

Reply:
Delineated the two major differences [here](https://github.com/tinker-michaelj/hiero-improvement-proposals/blob/main/HIP/hip-1195.md#0x16d-contract-address); everything else is the same as a contract transaction executed from top-level HAPI.